File: repos_estudiantes_ALVAREZ_primera_entrega/GARNICA_FUGUEROA_CAMILA_NATALIA/Polimorfismo/Ejercicio 1/1. Codigo/src/Ejercicio_1/Videojuego.java
package Ejercicio_1;

import java.util.ArrayList;

public class Videojuego {
	private String nombre;
	private String plataforma;
	private int cantJugadores;
	
	ArrayList<String> jugadores;
	
	//Constructor recibira solo el nombre del videojuego
	public Videojuego(String nombre) {
		this.nombre = nombre;
		this.plataforma = "Steam";
		this.cantJugadores = 1;
		this.jugadores = new ArrayList<>();
	}
	
	//constructor recbira la plataforma y el nro de jugadores
	public Videojuego(String nombre, String plataforma, int cantJugadores) {
		this.nombre = nombre;
		this.plataforma = plataforma;
		this.cantJugadores = cantJugadores;
		this.jugadores = new ArrayList<>();
	}
	
	public void mostrar () {
		System.out.println("El juego es "+ nombre + " de la plataforma " + plataforma + " con " + cantJugadores + " jugadores");
	}

	public void agregar(String jugador) {
		jugadores.add(jugador);
		System.out.println("Se añadio al jugador: " + jugador);
	}
	
	public void agregar(ArrayList<String> jugNuevos) {
		jugadores.addAll(jugNuevos);
		System.out.println("Se agregaron a los siguientes jugadores: "+ jugadores);
	}
}	

File: repos_estudiantes_ALVAREZ_primera_entrega/GARNICA_FUGUEROA_CAMILA_NATALIA/Polimorfismo/Ejercicio 1/1. Codigo/Videojuego.py
class Videojuego:
    def __init__(self, nombre, plataforma="Steam", cant_jugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cant_jugadores = cant_jugadores
        self.jugadores = []

    @classmethod
    def solo_nombre(cls, nombre):
        return cls(nombre)

    def mostrar(self):
        print(f"El juego es {self.nombre} de la plataforma {self.plataforma} con {self.cant_jugadores} jugadores")

    def agregar(self, jugador):
        if isinstance(jugador, list):
            self.jugadores.extend(jugador)
            print("Se agregaron a los siguientes jugadores:", self.jugadores)
        else:
            self.jugadores.append(jugador)
            print("Se añadió al jugador:", jugador)

print("SALA PARA JUGADORES EN SOLITARIO")

juego1 = Videojuego.solo_nombre("Brawl Stars")
juego1.mostrar()
juego1.agregar("Nataniel")

print()
print("SALA PARA MULTIPLES JUGADORES")

juego2 = Videojuego("Roblox", "Prime Gaming", 5)
juego2.mostrar()

jugadores = ["Lisa", "Belen", "Juliet"]
juego2.agregar(jugadores)
